netaddr.go

```diff
@@ -25,15 +25,13 @@ import (
 	"go4.org/intern"
 )
 
-// Simple constants copied to avoid importing math.
 const (
-	maxUint8  = 1<<8 - 1
+	// copied to avoid importing math.
 	maxUint16 = 1<<16 - 1
+	// mask4In6 is the mask of ip.lo indicating an IPv4-mapped IPv6 address.
+	mask4In6 = 0xffff00000000
```

Comment:
If this is the problematic bitmap that varies by endianness, the easiest fix is to put the different constants in build-tag-protected files. Or have an `IsLittleEndian` const that is 0 or 1, and do multiply-and-add tricks to construct the correct value from it (`IsLittleEndian*littleEndianBitmap+(1-IsLittleEndian)*bigEndianBitmap`). If you planned to use github.com/josharian/native, I'd be happy to add such a constant there. We could also move that to go4.org if we want everything in one place.

Or we could accept the BSWAPQ, of course.

Reply:
This isn't the problematic one, that's easy to address as you describe.

The hard one is the `mask6` helper function, which constructs the masks for prefix computation. If you use little-endian and end up with swapped around bytes in the uint64, the bits are no longer contiguous, and so you can't construct the mask with simple shifts. The closest I could get was to construct the mask with shifts and then insert a bswapq _there_, rather than at IP construction time. IMO that ends up being worse.

And also, I expect x86 to have optimized the stuffing out of "mov + bswapq", given how big-endian dominates networking, so I'm not particularly bothered. There are other wins we can extract from netaddr before we're worried about byte-swapping instructions in the compiled output :)